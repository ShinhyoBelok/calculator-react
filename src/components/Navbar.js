import React from 'react';
import { NavLink } from 'react-router-dom';
import '../css/navbar.css';

export default function Navbar() {
  return (
    <nav>
      <h1>Math Magicians</h1>
      <ul>
        <li>
          <NavLink
            to="/home"
            style={({ isActive }) => (isActive ? { color: 'orange' } : { color: 'white' })}
          >
            Home
          </NavLink>
        </li>
        <li>
          <NavLink
            to="/"
            style={({ isActive }) => (isActive ? { color: 'orange' } : { color: 'white' })}
          >
            Calculator
          </NavLink>
        </li>
        <li>
          <NavLink
            to="/quote"
            style={({ isActive }) => (isActive ? { color: 'orange' } : { color: 'white' })}
          >
            Quote
          </NavLink>
        </li>
      </ul>
    </nav>
  );
}
