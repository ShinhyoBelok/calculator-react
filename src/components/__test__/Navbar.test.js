import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../Navbar';

describe('Navbar component', () => {
  test('Test if Navbar render correctly', () => {
    const tree = renderer.create(<BrowserRouter><Navbar /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
