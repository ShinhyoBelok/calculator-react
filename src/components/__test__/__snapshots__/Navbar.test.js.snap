// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Test if Navbar render correctly 1`] = `
<nav>
  <h1>
    Math Magicians
  </h1>
  <ul>
    <li>
      <a
        className=""
        href="/home"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
          }
        }
      >
        Home
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "orange",
          }
        }
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="/quote"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
          }
        }
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
